### DISCUSSION TITLE
Support for Server Sent Events

### DISCUSSION THREAD

**Discussion by ekanna** [trillium-rs/trillium#40] [Ideas]
Please provide support for Server Sent Events (SSE)

https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events

**Comment by jbr** [comment]
What is your use case for server sent events that isn't met by websockets? I'm not opposed to building sse support but I want to understand the use case

**Reply by ekanna** [reply]
I prefer SSE because of its simplicity. Browsers automatically reconnects when ever connection is lost.

**Comment by jalcine** [comment]
My personal use-case for SSE is to use it as a read-only notification stream for clients. I'm working on an [IndieAuth](https://indieauth.spec.indieweb.org) server and being able to have a CSR app that can listen to a stream of info would be nice. This is definitely doable with WebSockets - one constraint is that this kind of thing is limited if it's not working over HTTP/2 (open connections issue)

**Comment by jbr** [comment]
[trillium-sse](https://docs.rs/trillium-sse/latest/trillium_sse/) now exists with support for stream-based server-sent events
